app/services/proofing/socure/id_plus/proofer.rb

```diff
@@ -77,9 +77,10 @@ def build_result_from_response(response)
         # @param [Proofing::Socure::IdPlus::Response] response
         # @return [Hash]
         def reason_codes_as_errors(response)
-          {
-            reason_codes: response.kyc_reason_codes.sort,
-          }
+          known_codes = SocureReasonCode.where(
+            code: response.kyc_reason_codes,
+          ).pluck(:code, :description).to_h
+          response.kyc_reason_codes.index_with { |code| known_codes[code] || '' }
```

Comment:
I like [Zach's suggestion below](https://github.com/18F/identity-idp/pull/11393/files#r1815487369) about making this `... || '[unknown]'` or something.